contrib/lib/src/templates/fairing.rs

```diff
@@ -0,0 +1,164 @@
+use super::DEFAULT_TEMPLATE_DIR;
+use super::context::Context;
+use super::engine::Engines;
+
+use rocket::Rocket;
+use rocket::config::ConfigError;
+use rocket::fairing::{Fairing, Info, Kind};
+
+#[cfg(not(debug_assertions))]
+mod context {
+    use std::ops::Deref;
+    use super::Context;
+
+    /// Wraps a Context. With `cfg(debug_assertions)` active, this structure
+    /// additionally provides a method to reload the context at runtime.
+    pub struct ContextManager(Context);
+
+    impl ContextManager {
+        pub fn new(ctxt: Context) -> ContextManager {
+            ContextManager(ctxt)
+        }
+
+        pub fn context<'a>(&'a self) -> impl Deref<Target=Context> + 'a {
+            &self.0
+        }
+    }
+}
+
+#[cfg(debug_assertions)]
+mod context {
+    extern crate notify;
+
+    use std::ops::{Deref, DerefMut};
+    use std::sync::{RwLock, Mutex};
+    use std::sync::mpsc::{channel, Receiver};
+
+    use super::{Context, Engines};
+
+    use self::notify::{raw_watcher, RawEvent, RecommendedWatcher, RecursiveMode, Watcher};
+
+    /// Wraps a Context. With `cfg(debug_assertions)` active, this structure
+    /// additionally provides a method to reload the context at runtime.
+    pub struct ContextManager {
+        /// The current template context, inside an RwLock so it can be updated
+        context: RwLock<Context>,
+        /// A filesystem watcher and the receive queue for its events
+        watcher: Option<(RecommendedWatcher, Mutex<Receiver<RawEvent>>)>,
+    }
+
+    impl ContextManager {
+        pub fn new(ctxt: Context) -> ContextManager {
+            let (tx, rx) = channel();
+
+            let watcher = if let Ok(mut watcher) = raw_watcher(tx) {
+                if watcher.watch(ctxt.root.clone(), RecursiveMode::Recursive).is_ok() {
+                    Some((watcher, Mutex::new(rx)))
+                } else {
+                    warn_!("Could not monitor the templates directory for changes.");
```

Comment:
Ah, so, the `_` just adds a level of indentation to allow for grouping messages. Here, we'd want the one on the top _not_ to use the `_` so we get this:

```rust
Warning: Could not monitor the templates directory for changes.
    => Live template reload will be unavailable.
```

In other words, the `_` is for cosmetic/grouping purposes.